### PR TITLE
Use the DNF manager to gather metadata

### PR DESCRIPTION
**Pull request by poncovka**
Reorganize the code that collects metadata about available packages and groups.

### CODE REVIEW

File: pyanaconda/payload/dnf/payload.py [hunk 1]
```diff
@@ -369,26 +369,6 @@ def get_addon_repo(self, repo_id):
...
         return repo
 
-    def _add_repo_to_dnf_and_ks(self, ksrepo):
-        """Add an enabled repo to dnf and kickstart repo lists.
-
-        Add the repo given by the pykickstart Repo object ksrepo to the
-        system.
-
-        Duplicate repos will not raise an error.  They should just silently
-        take the place of the previous value.
```
Comment (VladimirSlavik) [inline pyanaconda/payload/dnf/payload.py:389]: This is not exactly documented afterwards, but I guess you plan to refactor that further?

File: pyanaconda/payload/dnf/payload.py [hunk 2]
```diff
@@ -710,7 +674,7 @@ def update_base_repo(self, fallback=True, checkmount=True):
 
         self._include_additional_repositories()
         self._disable_unwanted_repositories()
-        self._load_enabled_repositories()
+        self._validate_enabled_repositories()
```
Comment (VladimirSlavik) [inline pyanaconda/payload/dnf/payload.py:677]: It seems repos are loaded when adding, and then here reloaded for validation? If so - how does the performance look? Are the metadata cached after first access, is cache enabled?

File: pyanaconda/payload/dnf/payload.py (hunk 1 above)
Comment (poncovka) [reply pyanaconda/payload/dnf/payload.py:389]: This is handled and documented in the `add_repository` method of the `DNFManager` class.

File: pyanaconda/modules/payloads/payload/dnf/dnf_manager.py [hunk 3]
```diff
@@ -906,8 +906,11 @@ def load_repository(self, repo_id):
         repo = self._get_repository(repo_id)
         url = repo.baseurl or repo.mirrorlist or repo.metalink
 
+        if not repo.enabled:
+            log.debug("Cannot load metadata from '%s'. The repo is disabled.", url)
```
[the remark this answers is not in the document]
Comment (poncovka) [reply pyanaconda/modules/payloads/payload/dnf/dnf_manager.py]: Fixed.

File: pyanaconda/payload/dnf/payload.py (hunk 2 above)
Comment (poncovka) [reply pyanaconda/payload/dnf/payload.py:677]: I have extended the documentation of `load_repository`.

File: pyanaconda/modules/payloads/payload/dnf/dnf_manager.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (VladimirSlavik) [reply pyanaconda/modules/payloads/payload/dnf/dnf_manager.py]: Sorry, what I had in mind was something like: `log.debug("Skipping loading metadata from disabled repo '%s'.", url)`

I don't want to keep the whole PR blocked on that, so acking anyway.

**Review by VladimirSlavik — APPROVED** [review]

**Pull request merged**